src/resolvers/__tests__/collection.test.js

```diff
@@ -2203,6 +2203,286 @@ describe('Collection', () => {
+            }`
+          }, {
+            contextValue
+          })
+
+          const { data } = response.body.singleResult
+
+          expect(data).toEqual({
+            collections: {
+              items: [{
+                conceptId: 'C100000-EDSC',
+                visualizations: {
+                  items: []
+                }
+              }, {
+                conceptId: 'C100001-EDSC',
+                visualizations: {
+                  items: []
+                }
+              }]
+            }
+          })
+        })
+      })
+
+      describe('associations are present in the metadata but not visualization associations', () => {
+        test('doesn\'t query for or return visualizations', async () => {
+          nock(/example-cmr/)
+            .defaultReplyHeaders({
+              'CMR-Took': 7,
+              'CMR-Request-Id': 'abcd-1234-efgh-5678'
+            })
+            .get(/collections\.json/)
+            .reply(200, {
+              feed: {
+                entry: [{
+                  id: 'C100000-EDSC',
+                  association_details: {
+                    variables: [{ concept_id: 'V100000-EDSC' }]
+                  }
+                }, {
+                  id: 'C100001-EDSC',
+                  association_details: {
+                    variables: [{ concept_id: 'V100000-EDSC' }]
+                  }
+                }]
+              }
+            })
+
+          const response = await server.executeOperation({
+            variables: {},
+            query: `{
+              collections {
+                items {
+                  conceptId
+                  visualizations {
+                    items {
+                      conceptId
+                    }
+                  }
+                }
+              }
+            }`
+          }, {
+            contextValue
+          })
+
+          const { data } = response.body.singleResult
+
+          expect(data).toEqual({
+            collections: {
+              items: [{
+                conceptId: 'C100000-EDSC',
+                visualizations: {
+                  items: []
+                }
+              }, {
+                conceptId: 'C100001-EDSC',
+                visualizations: {
+                  items: []
+                }
+              }]
+            }
+          })
+        })
+      })
+
+      describe('when visualization associations are present in the metadata', () => {
+        test('queries for and returns visualizations', async () => {
+          nock(/example-cmr/)
+            .defaultReplyHeaders({
+              'CMR-Took': 7,
+              'CMR-Request-Id': 'abcd-1234-efgh-5678'
+            })
+            .get(/collections\.json/)
+            .reply(200, {
+              feed: {
+                entry: [{
+                  id: 'C100000-EDSC',
+                  association_details: {
+                    visualizations: [{ concept_id: 'VIS100000-EDSC' }, { concept_id: 'VIS100001-EDSC' }]
```

Comment:
nit pick, but do you want these prefixes to be VISD?

Reply:
This is a published collection associated to a published visualization